frontend/src/components/TransactionLinks/TransactionLink.spec.js

```diff
@@ -44,115 +44,139 @@ describe('TransactionLink', () => {
       expect(wrapper.find('div.transaction-link').exists()).toBeTruthy()
     })
 
-    describe('Copy link to Clipboard', () => {
-      const navigatorClipboard = navigator.clipboard
-      beforeAll(() => {
-        delete navigator.clipboard
-        navigator.clipboard = { writeText: navigatorClipboardMock }
+    describe('Link validUntil Date is not valid', () => {
+      it('find no link copy button', () => {
+        expect(wrapper.find('.test-copy-link').exists()).toBe(false)
       })
-      afterAll(() => {
-        navigator.clipboard = navigatorClipboard
+
+      it('find no Qr-Code Button ', () => {
+        expect(wrapper.find('.test-qr-code').exists()).toBe(false)
       })
 
-      describe('copy with success', () => {
-        beforeEach(async () => {
-          navigatorClipboardMock.mockResolvedValue()
-          await wrapper.find('.test-copy-link').trigger('click')
-        })
-        it('should call clipboard.writeText', () => {
-          expect(navigator.clipboard.writeText).toHaveBeenCalledWith(
-            'http://localhost/redeem/c00000000c000000c0000',
-          )
-        })
-        it('toasts success message', () => {
-          expect(toastSuccessSpy).toBeCalledWith('gdd_per_link.link-copied')
-        })
+      it('find only Link Löschen Button ', () => {
+        expect(wrapper.find('.test-delete-link').exists()).toBe(true)
       })
     })
 
-    describe('qr code modal', () => {
-      let spy
-
-      beforeEach(() => {
+    describe('Link validUntil Date is valid ', () => {
+      beforeEach(async () => {
         jest.clearAllMocks()
+        await wrapper.setProps({
+          validUntil: '2022-05-30T14:22:40.000Z',
```

Comment:
```suggestion
          validUntil: new Date(now.getFullYear(), now.getMonth(), now.getDate() + 2),
```

Reply:
Bei deiner Lösung funktioniert der Test ab dem 31. Mai nicht mehr

Reply:
ok super, ich hatte vergeblich in nächtlichen versuchen ein new Date() erstellen wollen .. dann hatte ich es hart eingeschrieben.